pkg/proxy/handler.go

```diff
@@ -24,15 +27,33 @@ type Handler struct {
 	Queues      *queue.Manager
 }
 
+func NewHandler(deployments *deployments.Manager, endpoints *endpoints.Manager, queues *queue.Manager) *Handler {
+	return &Handler{Deployments: deployments, Endpoints: endpoints, Queues: queues}
+}
+
 func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var (
+		modelName          string
+		capturedStatusCode *int
```

Comment:
Why define `capturedStatusCode` as a pointer?

Reply:
This is pointing to the `statusCode` in `captureStatusResponseWriter`. But there is a simpler solution that may be more readable